third_party/amd/lib/TritonAMDGPUToLLVM/AsyncUtility.cpp

```diff
@@ -59,11 +60,16 @@ void addLocalBarrierAfterAmdGpuAsyncWait(ModuleOp mod) {
 
   IRRewriter builder(mod.getContext());
   for (auto waitOp : waits) {
-    if (isa<mlir::gpu::BarrierOp, gpu::LocalBarrierOp>(waitOp->getNextNode()))
+    if (isa<mlir::gpu::BarrierOp, amdgpu::MemoryCounterWaitOp>(
+            waitOp->getNextNode()))
       continue;
 
     builder.setInsertionPointAfter(waitOp);
-    builder.create<triton::gpu::LocalBarrierOp>(waitOp->getLoc());
+    auto dsAttr = builder.getI32IntegerAttr(0);
+    builder.create<amdgpu::MemoryCounterWaitOp>(
+        waitOp->getLoc(), /* load= */ nullptr, /* store= */ nullptr,
+        /* ds= */ dsAttr, /* exp= */ nullptr);
+    builder.create<mlir::ROCDL::SBarrierOp>(waitOp->getLoc());
```

Comment:
the gpu::localBarrierOp seems to be a good abstraction for dscnt(0)+s_barrier. Is there a reason to expand it?